cmd/mdatagen/metadata-schema.yaml

```diff
@@ -51,7 +51,7 @@ resource_attributes:
     # Optional: array of attribute values if they are static values (currently, only string type is supported).
     enum: [string]
     # Required: attribute value type.
-    type: <string|int|double|bool|bytes|slice|map>
+    type: <string|int|double|bool|bytes|slice|map|template>
```

Comment:
[According to the docs](https://github.com/open-telemetry/weaver/blob/main/schemas/semconv-syntax.md#template-type), the the value for a template type can be any type, but this implementation only accepts strings. Specifically, the type should be something like `template[type]`.

Could we add `template_key` a separate config option and use the `type` as the type for the value? I'm open to other approaches as well, but since the `template` type in semconv appears to apply to the value for the attribute key instead of the attribute value itself, I feel like it warrants a distinct config option.

Reply:
thanks for the feedback, adapted the code, please have a look if this solution fits your idea. Please keep in mind that only string is supported now, as functionality for other types will be added after you confirm

Reply:
should be adapted, please have a look